core/player/src/view/resolver/index.ts

```diff
@@ -167,10 +212,28 @@ export class Resolver {
     return updated.value;
   }
 
-  public getResolveCache() {
+  public getResolveCache(): Map<Node.Node, Resolve.ResolvedNode> {
     return new Map(this.resolveCache);
   }
 
+  /** Use flags to determine how to log the message */
+  private logCacheIssue(id: string, type: NodeType): void {
+    let message: string | undefined;
+    let loglevel: Severity | undefined;
+
+    if (type === NodeType.Asset || type === NodeType.View) {
+      loglevel = this.options.flagController.getFlag("duplicateIDLogLevel");
+      message = `Cache conflict: Found Asset/View nodes that have conflicting ids: ${id}, may cause cache issues.`;
+    } else if (type === NodeType.Value) {
+      loglevel = this.options.flagController.getFlag("cacheConflictLogLevel");
+      message = `Cache conflict: Found Value nodes that have conflicting ids: ${id}, may cause cache issues. To improve performance make value node IDs globally unique.`;
+    }
+
+    if (message && loglevel) {
+      this.logger?.[loglevel]?.(message);
+    }
+  }
```

Comment:
While most usages of flags I can think of right now are around logging but in theory some runtime functionality could depend on flags (like errors during expression parsing). Based on the current model a reference to the `FlagController` would have to be passed around but we could expose some dynamic logging functionality in the logger to take a flag and a message and let it internally figure out the severity to log with which would remove the need to pass around the reference as much.